src/request/swap-iframe/SwapIFrameApi.js

```diff
@@ -0,0 +1,283 @@
+        // }
+
+        // if (parsedRequest.fund.type === 'BTC' && request.fund.type === 'BTC') {
+        //     // Verify BTC HTLC address is correct from HTLC script
+        //     const givenAddress = parsedRequest.fund.recipientOutput.address;
+        //     const scriptAddress = BitcoinJS.payments.p2wsh({
+        //         // @ts-ignore Type 'Uint8Array' is not assignable to type 'Buffer'.
+        //         witness: [BitcoinJS.Buffer.from(request.fund.htlcScript)],
+        //         network: BitcoinUtils.Network,
+        //     }).address;
+
+        //     if (givenAddress !== scriptAddress) {
+        //         throw new Errors.InvalidRequestError('BTC output address does not match HTLC script');
+        //     }
+        // }
+
+        // if (parsedRequest.fund.type === 'NIM') {
+        //     // Check that validityStartHeight is before HTLC timeout
+        //     if (parsedRequest.fund.transaction.validityStartHeight >= parsedRequest.nimHtlc.timeoutBlockHeight) {
+        //         throw new Errors.InvalidRequestError(
+        //             'Fund validityStartHeight must be lower than HTLC timeout block height',
+        //         );
+        //     }
+        // }
+
+        // if (parsedRequest.redeem.type === 'NIM') {
+        //     // Check that validityStartHeight is before HTLC timeout
+        //     if (parsedRequest.redeem.transaction.validityStartHeight >= parsedRequest.nimHtlc.timeoutBlockHeight) {
+        //         throw new Errors.InvalidRequestError(
+        //             'Redeem validityStartHeight must be lower than HTLC timeout block height',
+        //         );
+        //     }
+        // }
+
+        // For BTC redeem transactions, the BitcoinJS lib validates that the output script of the input matches
+        // the witnessScript.
+
+        // TODO: Validate timeouts of the two contracts
+        // (Currently not possible because the NIM timeout is a block height, while the BTC timeout is a timestamp.
+        // And since we cannot trust the local device time to be accurate, and we don't have a reference for NIM blocks
+        // and their timestamps, we cannot compare the two.)
+        // When it becomes possible to compare (with Nimiq 2.0 Albatross), the redeem HTLC must have a higher timeout
+        // than the funding HTLC.
+
+        /** @type {KeyguardRequest.SignSwapTransactionsResult} */
+        const result = {};
+
+        if (request.fund.type === 'NIM') {
+            await loadNimiq();
+
+            const privateKey = new Nimiq.PrivateKey(Nimiq.BufferUtils.fromHex(privateKeys.nim));
+            const publicKey = Nimiq.PublicKey.derive(privateKey);
+
+            const transaction = this.parseTransaction({
+                ...request.fund,
+                flags: Nimiq.Transaction.Flag.CONTRACT_CREATION,
+                recipient: 'CONTRACT_CREATION',
+                recipientType: Nimiq.Account.Type.HTLC,
```

Comment:
Again, we force the three properties that we _know_ must be present for HTLC creation.